cmd/snap/main.go

```diff
@@ -22,18 +22,41 @@ package main
 import (
 	"fmt"
 	"os"
-	"text/tabwriter"
 
-	"github.com/ubuntu-core/snappy/client"
+	"github.com/ubuntu-core/snappy/logger"
+
+	"github.com/jessevdk/go-flags"
 )
 
-func main() {
-	cli := client.New()
+type options struct {
+	// No global options yet
+}
 
-	w := tabwriter.NewWriter(os.Stdout, 0, 8, 1, ' ', 0)
-	fmt.Fprintln(w, "What\tError\tResult")
-	sysInfo, err := cli.SysInfo()
-	fmt.Fprintf(w, "get system info\t%v\t%#v\n", err, sysInfo)
+var optionsData options
 
-	w.Flush()
+var parser = flags.NewParser(&optionsData, flags.HelpFlag|flags.PassDoubleDash)
+
+func init() {
+	err := logger.SimpleSetup()
+	if err != nil {
+		fmt.Fprintf(os.Stderr, "WARNING: failed to activate logging: %s\n", err)
+	}
+}
+
+func run() error {
+	_, err := parser.Parse()
+	if err != nil {
+		return err
+	}
+	if _, ok := err.(*flags.Error); !ok {
+		logger.Debugf("cannot parse arguments: %v: %v", os.Args, err)
+	}
+	return nil
+}
+
+func main() {
```

Comment:
As a very trivial nitpick, would you mind to move this up, right after `init()`? This is a trivial stub that won't grow much at all because all the meat will be inside `run()`, so moving it up we can see init, main, and run in a single screen right next to each other. `run` will grow longer, though.